packages/duckdb/index.cjs

```diff
@@ -185,17 +185,27 @@ module.exports.getRunner = async (opts, directory) => {
 	return async (queryContent, queryPath, batchSize) => {
 		// Filter out non-sql files
 		if (!queryPath.endsWith('.sql')) return null;
-		return runQuery(
-			queryContent,
-			{ ...opts, filename: path.join(directory, opts.filename) },
-			batchSize
-		);
+		// Determine filename based on string
+		const filename =
+			opts.filename.startsWith('md:?motherduck_token=') || opts.filename == ':memory:'
+				? opts.filename
+				: path.join(directory, opts.filename);
+		return runQuery(queryContent, { ...opts, filename: filename }, batchSize);
 	};
 };
 
 /** @type {import("@evidence-dev/db-commons").ConnectionTester<DuckDBOptions>} */
 module.exports.testConnection = async (opts, directory) => {
-	const r = await runQuery('SELECT 1;', { ...opts, filename: path.join(directory, opts.filename) })
+	// Determine filename based on string
+	const filename =
+		opts.filename.startsWith('md:?motherduck_token=') || opts.filename == ':memory:'
```

Comment:
a bit of copypasta here, should probably define elsewhere?

Reply:
Is there any reason this can't live in `runQuery`?